### ISSUE TITLE
Instance attribute becomes Unknown when type has __getattr__, but pyright handles it correctly

### ISSUE THREAD

**Issue by janfrederik** [DetachHead/basedpyright#1719]
### Description

When a class declares an instance attribute with an explicit type annotation, and that type is a class with `__getattr__`, basedpyright incorrectly reports the attribute as `Unknown` when accessed. Standard pyright does not have this issue.

The type is explicitly declared, and the assignment is type-checked. There's no code path where the attribute could be anything other than the declared type.

### Reproduction

```python
from __future__ import annotations
from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from gitlab.v4.objects import Project  # Has __getattr__ via RESTObject base class

class Handler:
    _project: Project  # Explicitly declared type

    def __init__(self, project: Project) -> None:
        self._project = project  # Assignment is type-checked

    def get_project(self) -> Project:
        return self._project  # Warning: Return type is unknown
```

[Playground link](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAKATgIYA2JApmADRQB2YAtrhPiVABQC2EpMlGnUYt2XXgKjCR4hFG69%2BUZq3Zde-QcNHipM%2BWgBcUMAGMwAZzBIYSTNh3dNvO8DQBtAPIArCBAAeYP4E-lqkPgC6UMr8oeEAKgQA1mAE2NAQoYkqALxQUABEcUkpMBraeoYFSirqmkUl5RoAvtX19Y1IZUA) - uses a simplified example since gitlab isn't available in playground.

### Expected Behavior

`self._project` should have type `Project` because:
1. The attribute is explicitly declared as `_project: Project`
2. The assignment `self._project = project` is type-checked to ensure type safety
3. There's no code path where `_project` could be anything other than `Project`

The `__getattr__` on `Project` is irrelevant - it affects attribute access **on** `Project` instances, not how other classes store `Project` as their own attribute.

### Actual Behavior

basedpyright reports:
```
warning: Return type is unknown (reportUnknownVariableType)
```

### Comparison with pyright

Standard pyright (tested with `pyright 1.1.x` via `uv run --with pyright pyright`) reports **no warnings** for the same code.

### Environment

- basedpyright 1.37.2
- Python 3.14
- The issue occurs with python-gitlab's `Project` class, which inherits from `RESTObject` that has `__getattr__(self, name: str) -> Any`

### Workaround

Using `__slots__` on the holder class makes basedpyright correctly recognize the type:

```python
class Handler:
    __slots__ = ("_project",)
    _project: Project
    # ... now _project is correctly typed as Project
```

But this shouldn't be necessary - the explicit type declaration should be sufficient.

**Comment by DetachHead** [comment]
> Standard pyright does not have this issue.

this is just because pyright's defaults are less strict than basedpyright's ([more info](https://docs.basedpyright.com/latest/benefits-over-pyright/better-defaults/#typecheckingmode)). the same error occurs in pyright if you enable these rules:

```toml
[tool.pyright]
reportUnknownMemberType = true
reportUnknownVariableType = true
```

> [Playground link](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAKATgIYA2JApmADRQB2YAtrhPiVABQC2EpMlGnUYt2XXgKjCR4hFG69%2BUZq3Zde-QcNHipM%2BWgBcUMAGMwAZzBIYSTNh3dNvO8DQBtAPIArCBAAeYP4E-lqkPgC6UMr8oeEAKgQA1mAE2NAQoYkqALxQUABEcUkpMBraeoYFSirqmkUl5RoAvtX19Y1IZUA) - uses a simplified example since gitlab isn't available in playground.

not sure what's going on here... i assume this isn't what you were trying to show:

<img width="1550" height="46" alt="Image" src="https://github.com/user-attachments/assets/168ad509-918b-4cdf-8599-39343176b754" />

> ### Reproduction
> ```py
> from __future__ import annotations
> from typing import TYPE_CHECKING, Any
> 
> if TYPE_CHECKING:
>     from gitlab.v4.objects import Project  # Has __getattr__ via RESTObject base class
> 
> class Handler:
>     _project: Project  # Explicitly declared type
> 
>     def __init__(self, project: Project) -> None:
>         self._project = project  # Assignment is type-checked
> 
>     def get_project(self) -> Project:
>         return self._project  # Warning: Return type is unknown
> ```
i'm not sure what was supposed to be in the minimal playground example, but in this example here the issue doesn't seem to be related to the `__getattr__`, but rather the `RepositoryMixin` class that `Project` extends:

```py
# gitlab/v4/objects/repositories.py
if TYPE_CHECKING:
    # When running mypy we use these as the base classes
    _RestObjectBase = gitlab.base.RESTObject
else:
    _RestObjectBase = object


class RepositoryMixin(_RestObjectBase):
    ...
```

it looks like some sort of hack to work around a mypy bug. but this isn't compatible with pyright because the `base` module cannot be accessed directly from `gitlab` that way ([more info](https://docs.basedpyright.com/latest/usage/import-statements/#loader-side-effects)). this causes the `_RestObjectBase` class to be `Unknown`, which is whatt's causing your error.

i would recommend opening an issue at https://github.com/python-gitlab/python-gitlab.

here's a minimal repro:

```py
class Foo(asdf):...  # error: reportUndefinedVariable

class Handler:
    _project: Foo

    def __init__(self, project: Foo) -> None:
        self._project = project

    def get_project(self):
        return self._project  # Warning: Return type is unknown
```

**Comment by janfrederik** [comment]
@DetachHead Ok, clear. Thank you for your analysis!

So I see a that just adding a simple `import gitlab.base` to the type checking block in `gitlab/v4/objects/repositories.py` does the trick:

```python
# gitlab/v4/objects/repositories.py
if TYPE_CHECKING:
    # When running mypy we use these as the base classes
    import gitlab.base         # Add this line
    _RestObjectBase = gitlab.base.RESTObject
else:
    _RestObjectBase = object


class RepositoryMixin(_RestObjectBase):
    ...
```

I'll file an issue with python-gitlab.
